File: skills/hook-creator/scripts/validate_hook.py
#!/usr/bin/env python3
"""Validate the hook component of a Goose/Open Plugins plugin."""

import argparse
from pathlib import Path

from hook_format import validate_hooks


def main():
    parser = argparse.ArgumentParser(description="Validate Goose plugin hooks")
    parser.add_argument("plugin_dir")
    args = parser.parse_args()
    root = Path(args.plugin_dir).expanduser().resolve()
    result = validate_hooks(root)
    for warning in result.warnings:
        print(f"WARNING: {warning}")
    if result.errors:
        for error in result.errors:
            print(f"ERROR: {error}")
        raise SystemExit(1)
    print(f"OK: {root / 'hooks' / 'hooks.json'}")


if __name__ == "__main__":
    main()

File: skills/hook-creator/scripts/validate_hook.ts
#!/usr/bin/env node
// Validate the hook component of a Goose/Open Plugins plugin.
import { resolve, join } from "node:path";
import { validateHooks } from "./hook_format.js";

function main() {
  const [pluginDir] = process.argv.slice(2);
  if (!pluginDir) {
    console.error("usage: validate_hook.js <plugin_dir>");
    process.exit(2);
  }
  const root = resolve(pluginDir);
  const result = validateHooks(root);
  for (const warning of result.warnings) {
    console.log(`WARNING: ${warning}`);
  }
  if (result.errors.length) {
    for (const error of result.errors) {
      console.log(`ERROR: ${error}`);
    }
    process.exit(1);
  }
  console.log(`OK: ${join(root, "hooks", "hooks.json")}`);
}

main();
